telegram/_poll.py

```diff
@@ -85,54 +86,82 @@ class PollAnswer(TelegramObject):
-            May be empty if the vote was retracted.
+        option_ids (Sequence[:obj:`int`]): Identifiers of answer options, chosen by the user. May
+            be empty if the user retracted their vote.
 
             .. versionchanged:: 20.0
                 |sequenceclassargs|
+        user (:class:`telegram.User`, optional): The user that changed the answer to the poll,
+            if the voter isn't anonymous.
 
-    Attributes:
-        poll_id (:obj:`str`): Unique poll identifier.
-        voter_chat (:class:`telegram.Chat`): The chat that changed the answer to the poll,
-            if the voter is anonymous.
+            .. versionchanged:: NEXT.VERSION
+                paramref:`user` became optional.
+        voter_chat (:class:`telegram.Chat`, optional): The chat that changed the answer to the
+            poll, if the voter is anonymous.
 
             .. versionadded:: NEXT.VERSION
-        user (:class:`telegram.User`): The user, who changed the answer to the poll,
-            if the voter isn't anonymous.
-        option_ids (Tuple[:obj:`int`]): Identifiers of answer options, chosen by the user.  May be
-            empty if the user retracted their vote.
+
+    Attributes:
+        poll_id (:obj:`str`): Unique poll identifier.
+        option_ids (Tuple[:obj:`int`]): Identifiers of answer options, chosen by the user. May
+            be empty if the user retracted their vote.
 
             .. versionchanged:: 20.0
                 |tupleclassattrs|
+        user (:class:`telegram.User`, optional): Optional. The user, who changed the answer to the
+            poll, if the voter isn't anonymous.
+
+            .. versionchanged:: NEXT.VERSION
+                :paramref:`user` became optional.
+        voter_chat (:class:`telegram.Chat`): Optional. The chat that changed the answer to the
+            poll, if the voter is anonymous.
+
+            .. versionadded:: NEXT.VERSION
 
     """
 
-    __slots__ = ("option_ids", "user", "poll_id", "voter_chat")
+    __slots__ = ("option_ids", "poll_id", "user", "voter_chat")
 
     def __init__(
         self,
         poll_id: str,
-        voter_chat: Chat,
-        user: User,
         option_ids: Sequence[int],
+        user: Optional[User],
+        voter_chat: Optional[Chat],
         *,
         api_kwargs: Optional[JSONDict] = None,
     ):
         super().__init__(api_kwargs=api_kwargs)
         self.poll_id: str = poll_id
-        self.voter_chat = voter_chat
-        self.user: User = user
-        self.option_ids: Tuple[int, ...] = parse_sequence_arg(option_ids)
-
-        self._id_attrs = (self.poll_id, self.voter_chat, self.user, tuple(self.option_ids))
+        self.voter_chat: Optional[Chat] = voter_chat
+
+        if isinstance(option_ids, User) or isinstance(user, Tuple):  # type: ignore[arg-type]
+            warn(
+                "From v20.5 the order of `option_ids` and `user` is changed as the later one"
```

Comment:
just for completeness: let's try to use "NEXT.VERSION" instead of fixed numbers :) should be okay for this PR though as there will surely be no other release in between :)